### DISCUSSION TITLE
Kernel Panic, unable to mount rootfs

### DISCUSSION THREAD

**Discussion by es-ggg** [polarfire-soc/polarfire-soc-documentation#382] [HSS and U-Boot]
[Kernel Panic Unable to mount rootfs v2023-09 BSP v2023-06 reference design .txt](https://github.com/polarfire-soc/polarfire-soc-documentation/files/13467532/Kernel.Panic.Unable.to.mount.rootfs.v2023-09.BSP.v2023-06.reference.design.txt)
Hello all,

I'm currently facing a kernel panic issue that prevents the icicle dev kit I have from booting. I updated the icicle kit reference design and HSS to v2023.06 using Flash Pro5. After successfully building the BSP v2023.09 and writing the .wic image to an SD card of size 64 GB. The dmesg error is in the attached text file. I had successfully built the BSP for v2023.09 before a couple of weeks back, and didn't experience this issue, but I deleted the build directory and rebuilt and have been saddled with this error since. Luckily, I saved that image from the previous build from a couple of weeks back, and re-tried it, and do not get this issue i.e. that .wic image boots successfully. I was wondering if anyone can point me to how to possibly disable the checksums that happen right when the icicle kit is trying to load the rootfs. i.e. "EXT4-fs (mmcblk0p3): ext4_check_descriptors: Checksum" in the dmesg logs. I didn't used to have those in the previous working image of the build. If this is something that cannot be done, can someone please direct me on how to check out v2023.06 of this repo https://github.com/polarfire-soc/meta-polarfire-soc-yocto-bsp/

For further information, I checked out and built the github repository using the google repo package 'repo'  like this:

$ mkdir yocto-dev && cd yocto-dev
$ repo init -u https://github.com/polarfire-soc/polarfire-soc-yocto-manifests.git -b main -m default.xml

repo sync
repo rebase

. ./meta-polarfire-soc-yocto-bsp/polarfire-soc_yocto_setup.sh
MACHINE=icicle-kit-es bitbake mpfs-dev-cli

And I wrote to the SD card like this: zcat mpfs-dev-cli-icicle-kit-es.wic.gz | sudo dd of=/dev/sdb bs=4096 iflag=fullblock oflag=direct conv=fsync status=progress

I was made aware that ideally the branches of the icicle reference design should match the version of the BSP. Therefore, since the latest version of the reference design that I've been able to get to work is v2023.06, which is the head of the repository at the time this question is being asked, I was wondering how to checkout the equivalent version v2023.06 of the BSP to appropriately pair them? I'm not very familiar with using the repo tool/package to checkout a specific version. When I attempted the commands "repo branch" and "repo branches" in my linux terminal, the result is "(no branches)" and when I try "repo checkout 2023.06", I get the following "repo checkout 2023.06
error: no project has branch 2023.06
Repo command failed: MissingBranchError"

I'm using Ubuntu 20.04 LTS (focal). I've been stuck on this for a couple of weeks, and it's preventing me from proceeding to modifying the BSP to add a new meta layer to automatically start the Software I want it to run after it loads the rootfs. Any help/clarification/information would be greatly appreciated.

Thanks,
noob user

**Comment by es-ggg** [comment]
After encountering the issue mentioned above, I decided to try these 2 things and they both solved my "problem" independently:

I attempted building the minimal image using the command: **MACHINE=icicle-kit-es bitbake core-image-minimal** and wrote the image to the SD like this: **sudo dd if=~/yocto-dev/build/tmp-glibc/deploy/images/icicle-kit-es/core-image-minimal-icicle-kit-es.wic of=/dev/sdb status=progress**

The steps listed above got the icicle dev kit to boot.

I then went on to also copy the previously built image '**mpfs-dev-cli-icicle-kit-es.rootfs.wic**' to my windows PC and used win32DiskImager app to write to another SD card, and that SD card also completed booting.

I'm not sure why the tarball '**mpfs-dev-cli-icicle-kit-es.wic.gz**' and the image '**mpfs-dev-cli-icicle-kit-es.rootfs.wic**' failed to properly write to the SD card (i.e. SD card failed to boot after successful write indication from Linux terminal) using the commands below when I used my Linux machine:

**zcat ~/yocto-dev/build/tmp-glibc/deploy/images/icicle-kit-es/mpfs-dev-cli-icicle-kit-es.wic.gz | sudo dd of=/dev/sdb bs=4096 iflag=fullblock oflag=direct conv=fsync status=progress**

**sudo dd if=~/yocto-dev/build/tmp-glibc/deploy/images/icicle-kit-es/mpfs-dev-cli-icicle-kit-es.wic of=/dev/sdb status=progress**

**Comment by rockoguido** [comment]
it seems you need to checkout the proper branch, that is 2023.06, when you obtain your repo.  use the following command...

repo init -u https://github.com/polarfire-soc/polarfire-soc-yocto-manifests.git -b 2023.06 -m default.xml

in this case you will NOT be using the main branch, instead you are using an earlier branch. 

I hope that helps

**Reply by es-ggg** [reply]
Thank you @rockoguido
